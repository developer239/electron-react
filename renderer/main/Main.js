import React, { Component } from 'react';
import styled from 'styled-components';

import { send } from '../_shared/messageHelper';


const Container = styled.div`
  padding: 0 20px;
`;

export default class Main extends Component {
  handleButtonClick() {
    send('open-settings-window');
  };

  render() {
    return (
      <Container>
        <h1>Hello World!</h1>
        <p>From minimal electron, es6, react application.</p>
        <h2>Counter</h2>
        <label>Current value: </label>value <br />
        <button onClick={this.handleButtonClick}>Open Counter</button>
      </Container>
    );
  }
}
