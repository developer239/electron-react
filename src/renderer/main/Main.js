import React, { Component } from 'react'
import { send, listenTo } from '../_shared/messageHelper'
import Button from '../_shared/components/Button'
import Container from '../_shared/components/Container'
import Content from '../_shared/components/Content'
import Header from '../_shared/components/Header'
import SubHeader from '../_shared/components/SubHeader'
import Text from '../_shared/components/Text'
import Value from '../_shared/components/Value'


export default class Main extends Component {
  constructor() {
    super()

    this.state = {
      counterValue: 0,
    }

    listenTo('counter-increment', () => {
      this.updateCounterValue(this.state.counterValue + 1)
    })

    listenTo('counter-decrement', () => {
      this.updateCounterValue(this.state.counterValue - 1)
    })

    listenTo('counter-set-value', (event, args) => {
      this.updateCounterValue(args.payload)
    })
  }

  updateCounterValue = counterValue => {
    this.setState({ counterValue })
  }

  handleButtonClick = () => send('open-counter-window')

  render() {
    const counterValue = this.state.counterValue

    return (
      <Container>
        <Header>Electron App</Header>
        <Content>
          <Text>Lorem ipsum dolor sit amet, nonumes convenire iracundia no sit, in possit audire cum, in duo mundi
            tantas dolorum. Graeco splendide mei ut. An his doming eirmod disputando, eum tritani meliore ne. Populo
            consequuntur nec cu, eos an accusata neglegentur interpretaris, enim ullamcorper deterruisset eu ius. Nec
            omnium dignissim mediocritatem ut. Possit aliquip deleniti vis ut, ut sea dicat augue adversarium.</Text>
          <SubHeader>Counter</SubHeader>
          Current value <Value>{counterValue}</Value><br />
          <Button onClick={this.handleButtonClick}>Open Counter</Button>
        </Content>
      </Container>
    )
  }
}
