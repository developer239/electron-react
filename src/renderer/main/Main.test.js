import React from 'react'
import sinon from 'sinon'
import { shallow } from 'enzyme'
import { Main } from './Main'


describe('Main Component', () => {
  const counterValue = 0
  const handleButtonClick = sinon.fake()

  it('renders with all props', () => {
    const wrapper = shallow((
      <Main
        counterValue={counterValue}
        handleButtonClick={handleButtonClick}
      />
    ))

    expect(wrapper).toMatchSnapshot()
  })

  it('it calls counter handlers', () => {
    const wrapper = shallow((
      <Main
        counterValue={counterValue}
        handleButtonClick={handleButtonClick}
      />
    ))
    const buttons = wrapper.find('Button')

    // Open Counter Window Handler
    buttons.at(0).simulate('click')
    expect(handleButtonClick.callCount).toBe(1)
  })
})
