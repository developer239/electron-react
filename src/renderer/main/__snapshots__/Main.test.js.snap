// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Component renders with all props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main
    counterValue={0}
    handleButtonClick={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header>
          Electron App
        </Header>,
        <Content>
          <P>
            Lorem ipsum dolor sit amet, nonumes convenire iracundia no sit, in possit audire cum, in duo mundi tantas dolorum. Graeco splendide mei ut. An his doming eirmod disputando, eum tritani meliore ne. Populo consequuntur nec cu, eos an accusata neglegentur interpretaris, enim ullamcorper deterruisset eu ius. Nec omnium dignissim mediocritatem ut. Possit aliquip deleniti vis ut, ut sea dicat augue adversarium.
          </P>
          <H2>
            Counter
          </H2>
          Current value 
          <Span>
            0
          </Span>
          <br />
          <Button
            onClick={[Function]}
          >
            Open Counter
          </Button>
        </Content>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Electron App",
        },
        "ref": null,
        "rendered": "Electron App",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <P>
              Lorem ipsum dolor sit amet, nonumes convenire iracundia no sit, in possit audire cum, in duo mundi tantas dolorum. Graeco splendide mei ut. An his doming eirmod disputando, eum tritani meliore ne. Populo consequuntur nec cu, eos an accusata neglegentur interpretaris, enim ullamcorper deterruisset eu ius. Nec omnium dignissim mediocritatem ut. Possit aliquip deleniti vis ut, ut sea dicat augue adversarium.
            </P>,
            <H2>
              Counter
            </H2>,
            "Current value ",
            <Span>
              0
            </Span>,
            <br />,
            <Button
              onClick={[Function]}
            >
              Open Counter
            </Button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Lorem ipsum dolor sit amet, nonumes convenire iracundia no sit, in possit audire cum, in duo mundi tantas dolorum. Graeco splendide mei ut. An his doming eirmod disputando, eum tritani meliore ne. Populo consequuntur nec cu, eos an accusata neglegentur interpretaris, enim ullamcorper deterruisset eu ius. Nec omnium dignissim mediocritatem ut. Possit aliquip deleniti vis ut, ut sea dicat augue adversarium.",
            },
            "ref": null,
            "rendered": "Lorem ipsum dolor sit amet, nonumes convenire iracundia no sit, in possit audire cum, in duo mundi tantas dolorum. Graeco splendide mei ut. An his doming eirmod disputando, eum tritani meliore ne. Populo consequuntur nec cu, eos an accusata neglegentur interpretaris, enim ullamcorper deterruisset eu ius. Nec omnium dignissim mediocritatem ut. Possit aliquip deleniti vis ut, ut sea dicat augue adversarium.",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Counter",
            },
            "ref": null,
            "rendered": "Counter",
            "type": [Function],
          },
          "Current value ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": 0,
            },
            "ref": null,
            "rendered": 0,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Open Counter",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Open Counter",
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header>
            Electron App
          </Header>,
          <Content>
            <P>
              Lorem ipsum dolor sit amet, nonumes convenire iracundia no sit, in possit audire cum, in duo mundi tantas dolorum. Graeco splendide mei ut. An his doming eirmod disputando, eum tritani meliore ne. Populo consequuntur nec cu, eos an accusata neglegentur interpretaris, enim ullamcorper deterruisset eu ius. Nec omnium dignissim mediocritatem ut. Possit aliquip deleniti vis ut, ut sea dicat augue adversarium.
            </P>
            <H2>
              Counter
            </H2>
            Current value 
            <Span>
              0
            </Span>
            <br />
            <Button
              onClick={[Function]}
            >
              Open Counter
            </Button>
          </Content>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Electron App",
          },
          "ref": null,
          "rendered": "Electron App",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <P>
                Lorem ipsum dolor sit amet, nonumes convenire iracundia no sit, in possit audire cum, in duo mundi tantas dolorum. Graeco splendide mei ut. An his doming eirmod disputando, eum tritani meliore ne. Populo consequuntur nec cu, eos an accusata neglegentur interpretaris, enim ullamcorper deterruisset eu ius. Nec omnium dignissim mediocritatem ut. Possit aliquip deleniti vis ut, ut sea dicat augue adversarium.
              </P>,
              <H2>
                Counter
              </H2>,
              "Current value ",
              <Span>
                0
              </Span>,
              <br />,
              <Button
                onClick={[Function]}
              >
                Open Counter
              </Button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Lorem ipsum dolor sit amet, nonumes convenire iracundia no sit, in possit audire cum, in duo mundi tantas dolorum. Graeco splendide mei ut. An his doming eirmod disputando, eum tritani meliore ne. Populo consequuntur nec cu, eos an accusata neglegentur interpretaris, enim ullamcorper deterruisset eu ius. Nec omnium dignissim mediocritatem ut. Possit aliquip deleniti vis ut, ut sea dicat augue adversarium.",
              },
              "ref": null,
              "rendered": "Lorem ipsum dolor sit amet, nonumes convenire iracundia no sit, in possit audire cum, in duo mundi tantas dolorum. Graeco splendide mei ut. An his doming eirmod disputando, eum tritani meliore ne. Populo consequuntur nec cu, eos an accusata neglegentur interpretaris, enim ullamcorper deterruisset eu ius. Nec omnium dignissim mediocritatem ut. Possit aliquip deleniti vis ut, ut sea dicat augue adversarium.",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Counter",
              },
              "ref": null,
              "rendered": "Counter",
              "type": [Function],
            },
            "Current value ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": 0,
              },
              "ref": null,
              "rendered": 0,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Open Counter",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Open Counter",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
